capa/ida/plugin/form.py

```diff
@@ -1037,7 +1000,7 @@ def update_rule_status(self, rule_text: str):
             return
 
         # we must create a deep copy of rules because any rule matching operations modify the original rule
-        rules = copy.deepcopy(self.ruleset_cache.rules)
+        rules = copy.deepcopy([r for r in self.ruleset_cache.rules.values() if not r.is_subscope_rule()])
```

Comment:
the is the one line (imho) that gets more complicated with the change to the ruleset cache change. let me know if it makes sense, needs doc, and/or should be considered for revert.

Reply:
I think some extra doc on this would be helpful. If it's used in other places (is it?) we could create a helper function.

Reply:
```suggestion
        # the ruleset may derive subscope rules from the source rules loaded from disk.
        # by ignoring them, we reconstruct the collection of rules provided by the user.
        rules = copy.deepcopy([r for r in self.ruleset_cache.rules.values() if not r.is_subscope_rule()])
```